src/dataflow/internal/process/functions/function-call.ts

```diff
@@ -104,6 +104,27 @@ export function processFunctionCall<OtherInfo>(functionCall: RFunctionCall<Other
 		inIds.push(...functionName.in, ...functionName.unknownReferences)
 	}
 
+	if(named && functionCallName == 'source') {
+		const sourceFile = functionCall.arguments[0]
+		if(sourceFile?.value?.type == RType.String) {
+			const request: RParseRequest = {
+				request:                'file',
+				content:                sourceFile.lexeme,
+				// TODO what does this do
+				ensurePackageInstalled: true
+			}
+			// TODO is this how to get a shell for this?
+			const shell = new RShell()
+			const slicer = new SteppingSlicer({
```

Comment:
For #615 to be useful for this, it looks like we might have to refactor and/or make a version of `SteppingSlicer` that uses the new synchronous shell. What do you think @EagleoutIce?

Reply:
or we defer that to the pipeline executor (with an issue) and, for now, do the steps ourselves.

Reply:
That doesn't appear possible, since some of the steps themselves require `RShell` and are, thus, inherently asynchronous. For example, `parse` and `normalize` both do.